aspnetcore/fundamentals/servers/kestrel.md

```diff
@@ -798,23 +803,40 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
 }
 ```
 
-The preceding middleware expects an `AllowedHosts` key in *appsettings.\<EnvironmentName>.json*. This key's value is a semicolon-delimited list of host names without the port numbers. Include the `AllowedHosts` key-value pair in *appsettings.Production.json*:
+The middleware expects an `AllowedHosts` key in *appsettings.json*/*appsettings.\<EnvironmentName>.json*. The value is a semicolon-delimited list of host names without port numbers:
 
-```json
-{
-  "AllowedHosts": "example.com"
-}
-```
+::: moniker-end
+
+::: moniker range=">= aspnetcore-2.1"
+
+As a workaround, use Host Filtering Middleware. Host Filtering Middleware is provided by the [Microsoft.AspNetCore.HostFiltering](https://www.nuget.org/packages/Microsoft.AspNetCore.HostFiltering) package, which is included in the [Microsoft.AspNetCore.App](xref:fundamentals/metapackage-app) metapackage. The middleware is added by [CreateDefaultBuilder](/dotnet/api/microsoft.aspnetcore.webhost.createdefaultbuilder), which calls [AddHostFiltering](/dotnet/api/microsoft.aspnetcore.builder.hostfilteringservicesextensions.addhostfiltering):
+
+[!code-csharp[](kestrel/samples-snapshot/2.x/KestrelSample/Program.cs?name=snippet_Program&highlight=9)]
+
+Host Filtering Middleware is disabled by default. To enable the middleware, define an `AllowedHosts` key in *appsettings.json*/*appsettings.\<EnvironmentName>.json*. The value is a semicolon-delimited list of host names without port numbers:
+
+::: moniker-end
 
-*appsettings.Development.json* (localhost configuration file):
+::: moniker range=">= aspnetcore-2.0"
+
+*appsettings.json*:
 
 ```json
 {
-  "AllowedHosts": "localhost"
+  "AllowedHosts": "example.com;localhost"
 }
 ```
 
+> [!NOTE]
+> Don't use Host Filtering Middleware and [Forwarded Headers Middleware](xref:host-and-deploy/proxy-load-balancer) ([ForwardedHeadersOptions.AllowedHosts](/dotnet/api/microsoft.aspnetcore.builder.forwardedheadersoptions.allowedhosts)) simultaneously. Host Filtering Middleware is appropriate when Kestrel is used as an edge server. Forwarded Headers Middleware is appropriate for reverse proxy server and load balancer host configurations.
```

Comment:
"Don't use simultaneously" is over-cautionary. "Note these middleware have similar functionality for different scenarios" is a more appropriate intro.

Reply:
You don't actually say here that the Forwarded Headers middleware also has an AllowedHosts option. Without that it's not obvious why you're mentioning that middleware.

Reply:
That's why I have it parenthetically present ...

> [Forwarded Headers Middleware](xref:host-and-deploy/proxy-load-balancer) ([ForwardedHeadersOptions.AllowedHosts](/dotnet/api/microsoft.aspnetcore.builder.forwardedheadersoptions.allowedhosts)) 

... but I'll add it as a phrase on the next commit and see if that helps.